certificate_loader.go

```diff
@@ -22,19 +22,21 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path/filepath"
+	"regexp"
 	"time"
 
 	"github.com/pavel-v-chernykh/keystore-go"
 	"golang.org/x/sys/unix"
 )
 
-const CACertificates = "/etc/ssl/certs/ca-certificates.crt"
+const DefaultCertsDir = "/etc/ssl/certs"
 
 type CertificateLoader struct {
-	CACertificatesPath string
-	KeyStorePassword   string
-	KeyStorePath       string
-	Logger             io.Writer
+	CertificateDirs  []string
```

Comment:
Digging into the openssl docs, it seems that we should support the case where both `SSL_CERT_FILE` and `SSL_CERT_DIR` are set.

https://www.openssl.org/docs/man1.1.0/man3/SSL_CTX_set_default_verify_paths.html